repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/write/LimitsResponseMimeTypeWriter.java

```diff
@@ -20,28 +20,44 @@
 package org.openrepose.filters.ratelimiting.write;
 
 import org.openrepose.filters.ratelimiting.util.LimitsEntityStreamTransformer;
+import org.springframework.http.MediaType;
 
-import javax.ws.rs.core.MediaType;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class LimitsResponseMimeTypeWriter {
 
+    public static final Set<MediaType> SUPPORTED_MEDIA_TYPES = Stream.of(
+            MediaType.APPLICATION_JSON,
+            MediaType.APPLICATION_XML
+            // TODO: Add support for MediaType.TEXT_XML
+    ).collect(Collectors.toSet());
+
     private final LimitsEntityStreamTransformer responseTransformer;
 
     public LimitsResponseMimeTypeWriter(LimitsEntityStreamTransformer responseTransformer) {
         this.responseTransformer = responseTransformer;
     }
 
     public MediaType writeLimitsResponse(byte[] readableContents, MediaType mediaType, OutputStream outputStream) throws IOException {
-        if (MediaType.APPLICATION_XML_TYPE.equals(mediaType)) {
+        /*
+         Check for JSON first since it is the preferred media type.
+         This is done explicitly even though JSON is also the default.
+          */
+        if (MediaType.APPLICATION_JSON.isCompatibleWith(mediaType)) {
+            responseTransformer.streamAsJson(new ByteArrayInputStream(readableContents), outputStream);
+            return MediaType.APPLICATION_JSON;
+        } else if (MediaType.APPLICATION_XML.isCompatibleWith(mediaType)) {
             outputStream.write(readableContents);
-            return MediaType.APPLICATION_XML_TYPE;
+            return MediaType.APPLICATION_XML;
         } else {
             // default to json for now
             responseTransformer.streamAsJson(new ByteArrayInputStream(readableContents), outputStream);
-            return MediaType.APPLICATION_JSON_TYPE;
+            return MediaType.APPLICATION_JSON;
```

Comment:
I hate that code is duplicated in the explicit JSON case and the default case. Technically, the default case should never be reached (at least, in rate limiting processing), but concerns are split between this class and the handler class, so that may not be clear. I couldn't come up with a quick solution given the methods available on `MediaType`, and I didn't want to do even more invasive refactoring, so this is how I left it.

I thought about extracting the JSON code to a method, but that's not much better. It certainly isn't very flexible is, say, we wanted to switch the default media type at some point.